### DISCUSSION TITLE
I ned help with search engines setting

### DISCUSSION THREAD

**Discussion by RomanAleynikov** [hoothin/SearchJumper#125] [Q&A] [answered]
Can I so tune extension it can to search for keywords on certain sites and  can I add this link on search pannel? I am sorry if my question is stupid.
![2024-05-12_181425](https://github.com/hoothin/SearchJumper/assets/87813216/09f9c361-bee2-4646-8bca-9912beacb424)

**Comment by hoothin** [comment]
https://github.com/hoothin/UserScripts/blob/master/SearchJumper/searchJumperDefaultConfig.js#L1346
```
https://www.google.com/search?q=%s%20site%3A%h&ie=utf-8&oe=utf-8
```
%h means hostname of your current viewing site.
y
You can change it to any domain you want.

**Reply by RomanAleynikov** [reply]
Thank you very much! Maybe is there more detail guide for Search Jumper? I read guide on the site after Search Jumper installing, but I am not native English so I understand not so good. I just don't want to disturb you my stupid questions in the future.

**Reply by hoothin** [reply]
I believe the 'search params' form in the 'search engines' tab of the configuration page covers all the functionality that I have been able to compile. Nonetheless, if you have any questions or concerns about any specific item, feel free to continue asking.

**Reply by RomanAleynikov** [reply]
Ok, thank you!  I have some questions more in this case.
1) Can I add in all engines page my own custom background image?
![2024-05-13_125531](https://github.com/hoothin/SearchJumper/assets/87813216/4c557fc8-e6e8-46b4-bea2-9855c036c1f4)

2) Can I change pop-up menu manually?
![2024-05-13_134546](https://github.com/hoothin/SearchJumper/assets/87813216/cbeb19a2-c572-40b7-970b-1bb1a30d4cf8)

3) Can I change right-click menu manually?
![2024-05-13_150440](https://github.com/hoothin/SearchJumper/assets/87813216/bace8606-1b28-4b54-ba9e-d05976f111e8)

4) I still don't understand where and how can I use parameters list to get any effect. Can I either use some parameter only in certain sites or any sites? Or must I know some programming language to use them? Is there some general pattern to use these parameters? Please can you show me using showTips parameter for example?
![2024-05-13_143043](https://github.com/hoothin/SearchJumper/assets/87813216/448e36de-1d0c-4123-a978-b0af58e41a75)

**Reply by hoothin** [reply, marked as answer]
1. At the moment, I haven't provided an option for custom backgrounds, but you can still customize the background using CSS. Here's how:
``` css
#search-jumper.search-jumper-showall {
    background-position: center 0px;
    background-repeat: no-repeat;
    background-size: cover;
    background-image: url(https://global.bing.com/th?id=OHR.CarlsbadNP_JA-JP0802484908_1920x1080.jpg)
}
```
Paste it into the "Custom css" on EXPORT CONFIG tab. Then click "Save".
2. Change the html by yourself.
![image](https://github.com/hoothin/SearchJumper/assets/5361213/748de1ec-6e79-47eb-8e2c-fae5b435a380)
3. These engines are based on your configuration and, in general, they are the first three within color-highlighted groups on your configuration page. For example, the 'Search in page' group.
4. The 'showTips' param is used to fetch information and display it according to a template. It can be used on any website. For specific details, please refer to the configuration of the IMDB rating.

**Reply by RomanAleynikov** [reply]
Hello! It's me again. I cannot understant what is these buttons meaning?
![image](https://github.com/hoothin/SearchJumper/assets/87813216/574885ce-0ed3-4117-8206-6606af4eb5db)

**Reply by hoothin** [reply]
It means the category will only be shown when a certain condition is met. For example, when you long press on an image, only the category with the 'Effect on image' option checked will be displayed.

**Reply by RomanAleynikov** [reply]
> It means the category will only be shown when a certain condition is met. For example, when you long press on an image, only the category with the 'Effect on image' option checked will be displayed.

Ok. Thank you!

**Reply by miluju** [reply]
> 1. At the moment, I haven't provided an option for custom backgrounds, but you can still customize the background using CSS. Here's how:
> 
> ```css
> #search-jumper.search-jumper-showall {
>     background-position: center 0px;
>     background-repeat: no-repeat;
>     background-size: cover;
>     background-image: url(https://global.bing.com/th?id=OHR.CarlsbadNP_JA-JP0802484908_1920x1080.jpg)
> }
> ```
> 
> Paste it into the "Custom css" on EXPORT CONFIG tab. Then click "Save". 2. Change the html by yourself. ![image](https://private-user-images.githubusercontent.com/5361213/330265971-748de1ec-6e79-47eb-8e2c-fae5b435a380.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc5NzU5NTgsIm5iZiI6MTc0Nzk3NTY1OCwicGF0aCI6Ii81MzYxMjEzLzMzMDI2NTk3MS03NDhkZTFlYy02ZTc5LTQ3ZWItOGUyYy1mYWU1YjQzNWEzODAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MjNUMDQ0NzM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDc3MDMyMjRlMGMxYmI2ZjI1YzBjNjQxYTFjNTg1MWU0ZDJkNTQyNzY3NDgyNmZlNzJmOGVjZGEyMDBkZWYyYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jW6NGAyZkNSWs8Pvrh-1iuql-q4kfoWg_vFzUYsNlXw) 3. These engines are based on your configuration and, in general, they are the first three within color-highlighted groups on your configuration page. For example, the 'Search in page' group. 4. The 'showTips' param is used to fetch information and display it according to a template. It can be used on any website. For specific details, please refer to the configuration of the IMDB rating.

**Comment by RomanAleynikov** [comment]
I have idea to translate this extension into Russian, but I haven't absolutely some experience in this deal. Can I to run extension locally on my computer or can I do some another way to immediately check translation chenges?

**Reply by hoothin** [reply]
Thank you! [Here](https://greasyfork.org/scripts/445274) is the link of Greasyfork version, you can install and modify the language object to check the changes. As for the configuration page, you can clone the entire project, then install Node.js, and double-click on start.cmd to check the changes.

**Reply by hoothin** [reply]
For example, you can add 
```
case 'ru':
case 'ru-RU':
  config = {
``` 
for Russian

**Reply by hoothin** [reply]
[[1](https://github.com/hoothin/SearchJumper/blob/main/searchJumper.user.js#L537)][[2](https://github.com/hoothin/SearchJumper/blob/main/src/index.js#L1040)] <- Here is the specific location for the PR. Once you've finished the modifications, you just need to sync these two files back to the repository.

**Reply by RomanAleynikov** [reply]
Please can you describe this more detaily? I dowloaded and instaled Node.js. I can edit .js files in any IDE, but I cannot check changes. I clicked double start.cmd but nothing happen. I mean configuration page file.

**Reply by hoothin** [reply]
Have you added your Node.js to the system environment variables? What would happen if you open the command prompt in the current folder and run `npm start`?

**Comment by RomanAleynikov** [comment]
I didn't do anything manualy after installing.
![image](https://github.com/hoothin/SearchJumper/assets/87813216/6a35a06a-1f24-4d0d-bc20-a4478a4b2794)

**Reply by hoothin** [reply]
Try running `npm install` to reinstall the dependencies.

**Reply by RomanAleynikov** [reply]
This is after reinstalling.
![image](https://github.com/hoothin/SearchJumper/assets/87813216/08058925-0a28-4bd1-9fbc-6eb199b79841)

**Reply by hoothin** [reply]
It seems that you run `npm start` in Desktop, you should move to the searchjumper folder and run it.

**Reply by RomanAleynikov** [reply]
There is error any way
![image](https://github.com/hoothin/SearchJumper/assets/87813216/ad33fe28-235b-4894-8749-65c02883bbd4)

**Reply by hoothin** [reply]
So did you run `npm install` in this specific folder? Maybe you ran it in a different directory like the Desktop?

**Reply by RomanAleynikov** [reply]
May be is there way to do the same using JetBrains IntellijIDEA? I have tryied to run local server. It is working.
![image](https://github.com/hoothin/SearchJumper/assets/87813216/67e96e80-b6cf-4bb8-999f-b5fc15389ced)

**Reply by hoothin** [reply]
Yes, I think so. [Running and debugging Node.js](https://www.jetbrains.com/help/idea/running-and-debugging-node-js.html)

**Reply by RomanAleynikov** [reply]
Probably I did wrong installing. You are right. Now installing is completed.
![image](https://github.com/hoothin/SearchJumper/assets/87813216/7f6ab2ae-561c-4c2f-a515-046c197d79b2)

**Reply by RomanAleynikov** [reply]
It seems it is working! Now can I change project files and the changes will display on localhos?
![image](https://github.com/hoothin/SearchJumper/assets/87813216/2ee1e3cc-5c7c-42d4-bde2-be0538afa4ab)

**Reply by hoothin** [reply]
Yes, Looking forward to your contribution!😀

**Reply by RomanAleynikov** [reply]
Ok. Thank you for supporting, and now I go to bed.

**Reply by RomanAleynikov** [reply]
So, it's time to ask questions)

1) I cannot understand this case meaning. How can I watch this case effect?
![image](https://github.com/hoothin/SearchJumper/assets/87813216/bdbc4eb2-66d7-443e-b1fd-f70283d083c7)

2) And here
![image](https://github.com/hoothin/SearchJumper/assets/87813216/4f082787-415e-4924-b4a2-746d15e1a4cc)

3) And here
![image](https://github.com/hoothin/SearchJumper/assets/87813216/71af8eb9-2399-4a65-aeb4-b57c60359c7d)

4) It seems this function don't work or I don't understand how it works. Check please this function. I tried different combinations and I couldn't get any result.
![image](https://github.com/hoothin/SearchJumper/assets/87813216/9b244b7d-76b3-4f27-b041-4218d7bffe9d)

5) I intuitively understand some of the functions here, but I can't reproduce them.
![image](https://github.com/hoothin/SearchJumper/assets/87813216/4b6f6e95-2f16-454a-b5df-d03994d4efe2)

6) Absolutely don't understand this function. After click on this button this menu is disappearing and nothing happen.
![image](https://github.com/hoothin/SearchJumper/assets/87813216/9230111b-b157-4228-8178-f9193dca9050)

**Reply by hoothin** [reply]
Good day! Let me explain.
1. You can call popup on input with `Ctrl + Right click` as you set before.
2. After you've added a search engine (like Google), when you perform a search on Google, a search bar will automatically appear on the side. This option determines whether the search bar should be fully expanded when it appears.
3. This setting refers to the engine that will be used to provide suggestions as you type in the search input box (the default hotkey is Alt+`, but this can be customized).
4. This option determines whether the search bar should automatically hide when you finish using it and move the mouse away from it.
5. Each search engine can have its own dedicated hotkey set individually. This option is the master toggle for all the hotkeys. Another option determines whether typing text in the input field should trigger the hotkeys. The third option specifies the hotkey to quickly cycle through the available search engines when you're already on a search engine page.
6. You can have multiple configuration pages, such as localhost, extension background page, and online one. This option is used to set the current configuration page as the default configuration page to be loaded when the menu is invoked.

**Reply by RomanAleynikov** [reply]
One more question. Is there any difference betwen category and group?
![image](https://github.com/hoothin/SearchJumper/assets/87813216/958d0dd8-7e85-4e50-8d62-2b584780f7a3)

**Reply by hoothin** [reply]
No, they are the same.

**Reply by RomanAleynikov** [reply]
Ok. And last questions for today.
1) Why these 3 groups are painted?
2) Why these 6 engines have yellow background color?
3) For what these 4 buttons need?
![image](https://github.com/hoothin/SearchJumper/assets/87813216/8c9b880a-6a74-4c3a-8ae0-b863764fa52a)

**Reply by hoothin** [reply]
No problem.
1. It's actually four painted groups, corresponding to the first text group, the first reverse image group, the first link group, and the first page group, respectively. The painted groups will be the default ones that expand when corresponding objects are invoked. For example, when you drag an image or `Ctrl+ right click` on an image, you will see 6 engines in the drag circle or an opened group. These are the first 6 engines in the first reverse image group you have configured. The text group and link group work the same way - when you invoke on selected text or a link, the corresponding group will appear.
2. When you click these 4 button, It will automatically jump to the corresponding group.

**Reply by hoothin** [reply]
![image](https://github.com/hoothin/SearchJumper/assets/5361213/adf8f392-c5fb-4c04-ae52-dfa2904b6e93)
You can unchecked the checkbox and then the color will disappear. This means it's no longer a special group that only appears when invoked on a specific object.

**Comment by RomanAleynikov** [comment]
Hi, it's me!

I have translated into Russian about 95% application's interface. I strove to translate the people can understand very simple. Most translation's part is very good, but there is little part which is not good. I have not translate search parameter list, because I little understand it and I don't want make translating mistake in this very important program's part. There is also little part which I didn't translate because I don't unserstant how this function works. If I will find some issues or add new translate or add rest part of translate I do it and write here. I can leave 2 files with my translation here. If you want add them in project after checking then do this. Or I can continue learning rest application functions and do more perfect translation.

P.s. I have added 2 commits and did synchronisation I think. Check it.

**Reply by hoothin** [reply]
Thank you! And you need to click pull request so I can merge it.
![image](https://github.com/hoothin/SearchJumper/assets/5361213/45eeb653-84b7-40fd-aca9-b4fb53e0f594)

**Reply by RomanAleynikov** [reply]
There is error

**Reply by RomanAleynikov** [reply]
![image](https://github.com/hoothin/SearchJumper/assets/87813216/774a6438-acd9-4707-8352-00df26de550a)

**Reply by hoothin** [reply]
I've already merged it into the online version, and once the store review is passed, it will be released. Thank you again for your contribution!

**Reply by RomanAleynikov** [reply]
I'm glad to help! Thank you for this usefull and powerfulll extension!
